CenterForOpenScience/osf.io
# [PLAT-767] Fix ordering for user nodes AND quickfiles

## Purpose

The order of nodes modified on the OSF dashboard is out of order when based on date modified. And it also fixes the user quickfiles view from being out of order.

## Changes

Dashboard:
- Simple fix for dashboard changes the ordering criteria to be correct.

Quickfiles:
- For the quickfiles it a far more complicated problem where the serialized order is different from the date modified order, so I've had to order it outside of a database query.
- Add test and modified tests to include file versions so `date_modified` appears.

## QA Notes

Dashboard:
I tested this by using the shell to change the date modified to be different then date last_logged to create the discrepancy that occurs normally. I've not sure how this could be forced to happen on staging, but recorded instances where it's known to happen should be fixed.

Quickfiles:
This bug happens because the “date modified” is the date that quickfiles are sorted by, but it’s not the date that appears to the user.

1. Upload a file to quickfiles
2. Upload a second file
3. Change the name of the second file, the date and place in order should remain unchanged (previously the date wouldn't change, but the order would)

## Documentation

Bug fix needs no docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/projects/PLAT/issues/PLAT-767
